src/twisted/python/compat.py

```diff
@@ -355,7 +359,7 @@ def nativeString(s: AnyStr) -> str:
     return s
 
 
-def _matchingString(constantString, inputString):
+def _matchingString(constantString: Union[bytes, str], inputString: AnyStr) -> AnyStr:
```

Comment:
On further consideration, this isn't the correct type for `_matchingString`. The `typing.AnyString` typevar can be bound to str or bytes, but not to `Union[str, bytes]`, which prevents `_matchingString` from typechecking if it's used in a function which must support both possibilities. I'll be pushing another commit which fixes this.